wallet/internal/db/queries/postgres/blocks.sql

```
@@ -1,12 +1,46 @@
 -- name: GetBlockByHeight :one
-SELECT block_height, header_hash, timestamp
+SELECT
+    block_height,
+    header_hash,
+    created_at
 FROM blocks
 WHERE block_height = $1;
 
 -- name: InsertBlock :exec
-INSERT INTO blocks (block_height, header_hash, timestamp)
-VALUES ($1, $2, $3);
+INSERT INTO blocks (block_height, header_hash, created_at)
+VALUES ($1, $2, $3)
+ON CONFLICT (block_height) DO UPDATE
+    SET
+        header_hash = excluded.header_hash,
+        created_at = excluded.created_at;
```

Is this intended to be an insert or an upsert? I wonder if it would be better to keep these operations separate and let an insert fail loudly when attempting to insert the same block twice. Ideally, the application should avoid this kind of race condition at the database layer, since allowing it could lead to an inconsistent state across goroutines.

---

Removed the `ON CONFLICT` part